// stream-chat-react-native-core/lib/commonjs/components/Message/MessageSimple/MessageTimestamp.js
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MessageTimestamp = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _ThemeContext = require("../../../contexts/themeContext/ThemeContext");
var _TranslationContext = require("../../../contexts/translationContext/TranslationContext");
var _getDateString = require("../../../utils/getDateString");
var _jsxRuntime = require("react/jsx-runtime");
var _this = this,
  _jsxFileName = "/home/runner/work/stream-chat-react-native/stream-chat-react-native/package/src/components/Message/MessageSimple/MessageTimestamp.tsx";
var MessageTimestamp = function MessageTimestamp(props) {
  var calendar = props.calendar,
    format = props.format,
    formatDate = props.formatDate,
    formattedDate = props.formattedDate,
    propsTDateTimeParser = props.tDateTimeParser,
    timestamp = props.timestamp;
  var _useTheme = (0, _ThemeContext.useTheme)(),
    _useTheme$theme = _useTheme.theme,
    grey = _useTheme$theme.colors.grey,
    timestampText = _useTheme$theme.messageSimple.content.timestampText;
  var _useTranslationContex = (0, _TranslationContext.useTranslationContext)(),
    contextTDateTimeParser = _useTranslationContex.tDateTimeParser;
  if (formattedDate) {
    return (0, _jsxRuntime.jsx)(_reactNative.Text, {
      style: [styles.text, {
        color: grey
      }, timestampText],
      children: formattedDate.toString()
    });
  }
  if (!timestamp) return null;
  var dateString = (0, _getDateString.getDateString)({
    calendar: calendar,
    date: timestamp,
    format: format,
    formatDate: formatDate,
    tDateTimeParser: propsTDateTimeParser || contextTDateTimeParser
  });
  if (!dateString) return null;
  return (0, _jsxRuntime.jsx)(_reactNative.Text, {
    style: [styles.text, {
      color: grey
    }, timestampText],
    children: dateString.toString()
  });
};
exports.MessageTimestamp = MessageTimestamp;
var styles = _reactNative.StyleSheet.create({
  text: {
    fontSize: 12
  }
});
